### PR TITLE
vmm: Remove all references to 'mmio' feature

### LINKED ISSUE #1843: Remove virtio MMIO transport layer
Since we have very few (none?) use cases for non PCI platforms, and supporting both PCI and MMIO as transport layers adds both code and testing/CI complexity, we could remove that feature from the code base.

/cc @cloud-hypervisor/architecture-committee

### PR DESCRIPTION
**Pull request by sboeuf**
This pull request removes all references to virtio-mmio. This means integration tests no longer support the mmio feature, and all the CI scripts only build for `virtio-pci`. This will reduce the time spent by the CI on every PR, as well as reducing the complexity of the codebase.

Fixes #1843

### CODE REVIEW

**Comment by sboeuf** [comment]
This is saving almost 4 minutes out of 18 minutes we were spending on the integration tests.

**Review by michael2012z — CHANGES_REQUESTED** [review]
There is a bit more to update in docs. In [device_model.md](https://github.com/cloud-hypervisor/cloud-hypervisor/blob/master/docs/device_model.md#virtio-devices) we mentioned the support of virtio-mmio which should be removed now.

In docs/arm64.md MMIO is also mentioned. I will create a PR to do some clean-up related to virtio-mmio on arm64 and update the doc as well, after this PR merged.

**Comment by sboeuf** [comment]
@michael2012z 
>There is a bit more to update in docs. In device_model.md we mentioned the support of virtio-mmio which should be removed now.
>In docs/arm64.md MMIO is also mentioned. I will create a PR to do some clean-up related to virtio-mmio on arm64 and update the doc as well, after this PR merged.

Let's do everything properly in this PR. Could you push more commits to my branch directly?

**Comment by michael2012z** [comment]
@sboeuf I can try to add some commits during the weekend, but cannot guarantee.  Just a little bit busy in recent days.

**Comment by sboeuf** [comment]
@michael2012z 
>I can try to add some commits during the weekend, but cannot guarantee. Just a little bit busy in recent days.

No worries, I want to wait for @rbradford feedback anyway (he'll be back next week).

**Comment by sboeuf** [comment]
@michael2012z I've updated the documentation. Feel free to add more.

**Comment by michael2012z** [comment]
@sboeuf I pushed 1 more commit to your branch to remove GICv2 from AArch64.

**Comment by sboeuf** [comment]
@michael2012z looks good, thanks!

**Comment by rbradford** [comment]
Looks good.

File: vmm/Cargo.toml
Comment (rbradford) [inline vmm/Cargo.toml:8]: This and the other similar cases should just be made dependencies.

**Review by michael2012z — APPROVED** [review]

File: vmm/Cargo.toml
Comment (sboeuf) [reply vmm/Cargo.toml]: I've updated the PR to address this comment. PTAL.

**Review by rbradford — APPROVED** [review]
Mostly a rslgtm as it building is quite sufficient for testing. More simplification of e.g. device_manager.rs might be forthcoming.

**Pull request merged**
